### ISSUE TITLE
Impossible to deploy - fibers@1.0.5 error

### ISSUE THREAD

**Issue by mryraghi** [arunoda/meteor-up-legacy#693]
I really don't know what to do. It's been three days that I've been trying to deploy my Meteor app to a Digital Ocean droplet. IMPOSSIBLE. Neither MUP nor any normal/standard way to deploy an app work.
I've deleted and reconfigured something like 5-6 droplets.

mup deploy:

```
x Invoking deployment process: FAILED

    -----------------------------------STDERR-----------------------------------
    ode_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
    gyp ERR! cwd /opt/snowdays/tmp/bundle/programs/server/node_modules/fibers
    gyp ERR! node -v v4.0.0
    gyp ERR! node-gyp -v v3.0.1
    gyp ERR! not ok
    Build failed
    npm ERR! Linux 3.13.0-57-generic
    npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install"
    npm ERR! node v4.0.0
    npm ERR! npm  v2.14.2
    npm ERR! code ELIFECYCLE

    npm ERR! fibers@1.0.5 install: `node ./build.js`
    npm ERR! Exit status 1
    npm ERR!
    npm ERR! Failed at the fibers@1.0.5 install script 'node ./build.js'.
    npm ERR! This is most likely a problem with the fibers package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node ./build.js
    npm ERR! You can get their info via:
    npm ERR!     npm owner ls fibers
    npm ERR! There is likely additional logging output above.

    npm ERR! Please include the following file with any support request:
    npm ERR!     /opt/snowdays/tmp/bundle/programs/server/npm-debug.log
    -----------------------------------STDOUT-----------------------------------
    nary npm modules
    bindings@1.0.0 node_modules/bindings

    nodeunit@0.9.1 node_modules/nodeunit
    └── tap@0.7.1 (inherits@2.0.1, buffer-equal@0.0.1, deep-equal@1.0.1, slide@1.1.6, yamlish@0.0.7, nopt@3.0.4, mkdirp@0.5.1, difflet@0.2.6, glob@4.5.3, runforcover@0.0.2)
    make: Entering directory `/opt/snowdays/tmp/bundle/programs/server/npm/npm-bcrypt/node_modules/bcrypt/build'
      CXX(target) Release/obj.target/bcrypt_lib/src/blowfish.o
      CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt.o
      CXX(target) Release/obj.target/bcrypt_lib/src/bcrypt_node.o
    make: Leaving directory `/opt/snowdays/tmp/bundle/programs/server/npm/npm-bcrypt/node_modules/bcrypt/build'

    > fibers@1.0.5 install /opt/snowdays/tmp/bundle/programs/server/node_modules/fibers
    > node ./build.js

    make: Entering directory `/opt/snowdays/tmp/bundle/programs/server/node_modules/fibers/build'
      CXX(target) Release/obj.target/fibers/src/fibers.o
    make: Leaving directory `/opt/snowdays/tmp/bundle/programs/server/node_modules/fibers/build'
```

**fibers** seems to be the cause of problems. 

Please help me :+1:

**Comment by MasterJames** [comment]
Please try the alternate dev branch mupx first.
If you're still having a problem please post your mup settings and (single run) logs if different, and any other environment info like OS versions etc.

**Comment by mryraghi** [comment]
That's my first problem using mupx for the first time. Any suggestion?

```
x Installing Docker: FAILED

    -----------------------------------STDERR-----------------------------------
    bash: line 4: docker: command not found
    sudo: no tty present and no askpass program specified
    sudo: no tty present and no askpass program specified
    sudo: no tty present and no askpass program specified
    sudo: no tty present and no askpass program specified
    -----------------------------------STDOUT-----------------------------------
    ----------------------------------------------------------------------------
```

Thank you!

**Comment by mryraghi** [comment]
This is my mup.json

```
{
  // Server authentication info
  "servers": [
    {
      "host": "<myIP>",
      "username": "<myusername>",
      "password": "<mypassord>",
      // or pem file (ssh based authentication)
      // WARNING: Keys protected by a passphrase are not supported
      //"pem": "~/.ssh/id_rsa"
      // Also, for non-standard ssh port use this
      //"sshOptions": { "port" : 49154 },
      // server specific environment variables
      "env": {}
    }
  ],

  // Install MongoDB on the server. Does not destroy the local MongoDB on future setups
  "setupMongo": true,

  // Application name (no spaces).
  "appName": "snowdays",

  // Location of app (local directory). This can reference '~' as the users home directory.
  // i.e., "app": "~/Meteor/my-app",
  // This is the same as the line below.
  "app": "/Users/Romeo/WebstormProjects/Other/snowdays",

  // Configure environment
  // ROOT_URL must be set to your correct domain (https or http)
  "env": {
    "PORT": 80,
    "ROOT_URL": "http://<myIP>"
  },

  // Meteor Up checks if the app comes online just after the deployment.
  // Before mup checks that, it will wait for the number of seconds configured below.
  "deployCheckWaitTime": 120,

  // show a progress bar while uploading.
  // Make it false when you deploy using a CI box.
  "enableUploadProgressBar": true
}

```

**Comment by MasterJames** [comment]
On Ubuntu this is recommended...

```
sudo apt-get install docker.io
```

You'd have to customize your remote instance.
https://coreos.com/os/docs/latest/customizing-docker.html

Anyway I'm seeing now what appears to be the main problem.
You're on an unsupported platform. Linux 3.13.0-57-generic should be Ubuntu 14.
Correct me if I'm wrong. So the questioning then leads to what service you are deploying on and if you can spin up an Ubuntu 14 instance?
This looks like the right kind of thing...
https://aws.amazon.com/marketplace/pp/B00JV9JBDS

**Comment by MasterJames** [comment]
There's also a mupx migration guide over here to.
https://github.com/arunoda/meteor-up/tree/mupx#migrating-from-meteor-up-0x

**Comment by mryraghi** [comment]
Thank you, I used AWS and it worked ;)

**Comment by demiters** [comment]
I'm having the same issue, don't think moving to AWS is a solution?

**Comment by mryraghi** [comment]
I just created another droplet without configuring it with a SSH-key-based authentication. Just username and password. It's working perfectly ;)

**Comment by MasterJames** [comment]
Okay great congratulations!
Here is something about the subject of PEM and SSH at Digital Ocean maybe you missed something simple there.
https://www.digitalocean.com/community/tutorials/how-to-connect-to-your-droplet-with-ssh
It's more secure if you can or make a robust 12 character long password.

**Comment by demiters** [comment]
How is this an issue of PEM/SSH if the deployment fails halfway through?

Also, I was advised to change nodeVersion from latest stable to 0.10.40 in mup.json and that actually worked and the app is now deployed. But why did it work?

**Comment by MasterJames** [comment]
It was a question of isolating the problem. The original poster stopping using it so logically it made sense there was something missing in their setup and so a link was provided for their problem. Looks like we have our threads crossed.
The original error message also says...

```
Please include the following file with any support request:
npm ERR!         /opt/snowdays/tmp/bundle/programs/server/npm-debug.log
```

... so the more information the more informed an answer one can expect or at least sometime a better place to start isolating the fault.
Anyway it looks like 0.10.40 had some nessecary improvement compatibilties with fibers etc. It's not worth exploring. One can always hope for better error messages and self diagnostics repair etc.

**Comment by sauliuni** [comment]
Same problem here. Using:

Ubuntu@12.04
npm@3.3.12
node@v5.3.0

npm-debug.log shows following:

```
1545 warn EPACKAGEJSON meteor-dev-bundle@0.0.0 No description
1546 warn EPACKAGEJSON meteor-dev-bundle@0.0.0 No repository field.
1547 warn EPACKAGEJSON meteor-dev-bundle@0.0.0 No license field.
1548 verbose stack Error: fibers@1.0.5 install: `node ./build.js`
1548 verbose stack Exit status 1
1548 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/lib/utils/lifecycle.js:232:16)
1548 verbose stack     at emitTwo (events.js:87:13)
1548 verbose stack     at EventEmitter.emit (events.js:172:7)
1548 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/lib/utils/spawn.js:24:14)
1548 verbose stack     at emitTwo (events.js:87:13)
1548 verbose stack     at ChildProcess.emit (events.js:172:7)
1548 verbose stack     at maybeClose (internal/child_process.js:818:16)
1548 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
1549 verbose pkgid fibers@1.0.5
1550 verbose cwd /opt/meteor/tmp/bundle/programs/server
1551 error Linux 3.8.0-42-generic
1552 error argv "/usr/bin/nodejs" "/usr/bin/npm" "install"
1553 error node v5.3.0
1554 error npm  v3.3.12
1555 error code ELIFECYCLE
1556 error fibers@1.0.5 install: `node ./build.js`
1556 error Exit status 1
1557 error Failed at the fibers@1.0.5 install script 'node ./build.js'.
1557 error Make sure you have the latest version of node.js and npm installed.
1557 error If you do, this is most likely a problem with the fibers package,
1557 error not with npm itself.
1557 error Tell the author that this fails on your system:
1557 error     node ./build.js
1557 error You can get their info via:
1557 error     npm owner ls fibers
1557 error There is likely additional logging output above.
1558 verbose exit [ 1, true ]
```

I wonder why the fiber 1.0.5 is used, since 1.0.8 is the latest?

**Comment by MasterJames** [comment]
Ubuntu 14 is the only officially supported deployment target.

**Comment by dhl** [comment]
If you are using Ubuntu 14.04, apt-get install nodejs, nodejs-legacy and npm should do the trick. The versions from NodeSource didn't work for me either, and results in the failure when building fibers@1.0.5.

**Comment by JeremyIglehart** [comment]
I was having this same problem the other day - I waited a while and tried again to "mupx deploy" and then it just worked.

Today, I had the problem again.  But then I found this page - and I followed [this suggestion](https://github.com/arunoda/meteor-up/issues/693#issuecomment-180443987) by @dhl and then "mupx deployed" again - and it seams to work fine now.  Hopefully that will get rid of my problem.

I didn't realize docker will use a nodejs installation on the server, not in the docker instance.  I'm still figuring this docker stuff out.

Thanks @dhl !
